service_container/tags.rst

```diff
@@ -564,3 +564,242 @@ application handlers.
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" index_attribute_name="handler_one" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" index_attribute_name="handler_two" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged" tag="app.handler" index-by="index_attribute_name" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+
+        $container->register(App\Handler\One::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'handler_one']);
+
+        $container->register(App\Handler\Two::class)
+            ->addTag('app.handler', ['index_attribute_name' => 'handler_two']);
+
+        $container->register(App\Handler\HandlerCollection::class)
+            // inject all services tagged with app.handler as first argument
+            ->addArgument(new TaggedIteratorArgument('app.handler', 'index_attribute_name'));
+
+After compilation the ``HandlerCollection`` service is able to iterate over your
+application handlers. But we want to retrieve a specific service by it's
+``index_attribute_name``. As it's an iterator, we can use ``iterator_to_array``
+to get an array and then retrieve the ``handler_two`` handler :
+
+.. code-block:: php
+
+    // src/Handler/HandlerCollection.php
+    namespace App\Handler;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlers)
+        {
+            $handlers = iterator_to_array($handlers);
+
+            $handler_two = $handlers['handler_two']:
+            $handler_one = $handlers['handler_one'];
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, by implementing a static
+    method named like this : ``getDefaultIndexAttributeNameName``
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            services:
+                App\Handler\One:
+                    tags:
+                        - { name: 'app.handler', priority: 20 }
+
+                App\Handler\Two:
+                    tags:
+                        - { name: 'app.handler', index: 'handler_two' }
+
+                App\HandlerCollection:
+                    # inject all services tagged with app.handler as first argument
+                    arguments: [!tagged { tag: 'app.handler', index_by: 'index' }]
```

Comment:
To make this shorter, we can "continue" the example from above. Basically, just show the `App\Handler\One` service. then a `...` to indicate that Two and HandlerCollection are the same.

Reply:
ok thanks